tests/api_tests/users/test_views.py

```diff
@@ -203,6 +206,241 @@ def test_get_projects_logged_in_as_different_user(self):
+                'title': '',
+                'startMonth': 1,
+                'endMonth': None,
+                'endYear': None,
+                'ongoing': False,
+                'department': '',
+                'institution': 'Waffle House'
+            }
+        ]
+        self.user_one.save()
+        self.auth_one = (self.user_one.username, 'justapoorboy')
+        self.user_one_url = "/v2/users/{}/".format(self.user_one._id)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('justapoorboy')
+        self.user_two.save()
+        self.auth_two = (self.user_two.username, 'justapoorboy')
+
+        self.new_fullname = 'el-Hajj Malik el-Shabazz'
+        self.new_given_name = 'Malcolm'
+        self.new_family_name = 'X'
+        self.new_suffix = 'Sr.'
+        self.new_employment_institutions = [
+            {
+                'startYear': '1982',
+                'title': '',
+                'startMonth': 1,
+                'endMonth': 4,
+                'endYear': 1999,
+                'ongoing': True,
+                'department': 'department of revolution',
+                'institution': 'IHop'
+            }
+        ]
+
+        self.new_educational_institutions = [
+            {
+                "startYear": '',
+                "degree": '',
+                "startMonth": None,
+                "endMonth": None,
+                "endYear": '2000',
+                "ongoing": False,
+                "department": 'Mom',
+                "institution": 'Heeyyyy'
+            }
+        ]
+
+        self.newGithub = 'newGithub'
+        self.newScholar = 'newScholar'
+        self.newPersonal_website = 'http://www.newpersonalwebsite.com'
+        self.newTwitter = 'newTwitter'
+        self.newLinkedIn = 'newLinkedIn'
+        self.newImpactStory = 'newImpactStory'
+        self.newOrcid = 'newOrcid'
+        self.newResearcherId = 'newResearcherId'
+
+    def test_patch_user_logged_out(self):
+        res = self.app.patch_json(self.user_one_url, {
+            'fullname': self.new_fullname,
+        }, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_patch_user_read_only_field(self):
+        # Logged in user updates their user information via patch
+        res = self.app.patch_json(self.user_one_url, {
+            'employment_institutions': self.new_employment_institutions,
+            'educational_institutions': self.new_educational_institutions,
+            'fullname': self.new_fullname,
+
+        }, auth=self.auth_one)
+        assert_equal(res.status_code, 200)
+        assert_not_equal(res.json['data']['employment_institutions'], self.new_employment_institutions)
```

Comment:
This could be a bit more specific. I'd rather do a positive test, especially if you're comparing a dictionary, so that it's not passing just because there's an odd quirk in the comparison between the json and the dictionary or similar. Making sure it's still equal to the old version would be great.